OpenRA.Mods.Cnc/Traits/Render/WithBuildingBib.cs

```diff
@@ -56,7 +56,7 @@ public IEnumerable<IActorPreview> RenderPreviewSprites(ActorPreviewInitializer i
 
 				// Some mods may define terrain-specific bibs
 				var sequence = Sequence;
-				if (map.Tiles.Contains(cell))
+				if (((IMapTiles)map).Tiles.Contains(cell))
```

Comment:
to me it seems more logical to let a map own a TileManager. i.e. map.GetTileManager(). promoting splitting logic.

i.e. otherwise you could have a map implement endless interfaces.

Reply:
Map no longer has `Tiles`, this functionality now belongs to `IMapTiles` and the current implementation sits in `DefaultMap` rather than `Map`. And `Tiles` is a CellLayer class which seems to work the same as your proposed `TileManager` class. In any case changing how Tiles are managed is not the goal of this PR. All I am doing here is moving the functionality out of `Map.cs` and into `DefaultMap`, behind an interface `IMapTiles`.

Reply:
i understand. but then still, it does not make sense to have IMapTiles inherit from IMap. Completely different things. Like letting ITrait inherit from IWidget. just because some class will implement both.

but i understand how you came to this solution.

Reply:
I agree, but this is how it was done with the Tilesets refactoring as well (see below link), with `ITemplatedTerrainInfo` and `ITerrainInfoNotifyMapCreated` both inheriting from `ITerrainInfo`. The downside is it adds a dependency to `IMap` where it may not need to be there, but on the plus side it means less casting is necessary, since you can cast to `IMapTiles` and still use the functionality from `IMap`, rather than having to cast twice.

https://github.com/OpenRA/OpenRA/pull/18728/files#diff-b08312ffb1592016a1b6797006dca462f3eb0f22a38ca2956e55e1e9c4f49ee1R14-R23